runtime/jvm/src/main/kotlin/kotlinx/serialization/Resolving.kt

```diff
@@ -67,7 +68,12 @@ fun serializerByTypeToken(type: Type): KSerializer<Any> = when(type) {
     }
     is ParameterizedType -> {
         val rootClass = (type.rawType as Class<*>)
-        val args = (type.actualTypeArguments)
+        val args = (type.actualTypeArguments).map { argument ->
+            when (argument) {
+                is WildcardType -> argument.upperBounds.first() as Class<*>
```

Comment:
Are you sure that this cast will always succeed? can't upper bound be `ParameterizedType` or `WildcardType` too?

Reply:
no i am not, added more typechacked and fallback to the uncasted argument, that might fail later, but adding branches should be trivial now